File: exercism/python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")

    result = number
    steps = 0
    while result > 1:
        steps += 1
        if result % 2 == 0:
            result = int(result / 2)
        else:
            result = 3 * result + 1
    
    return steps

File: exercism/typescript/collatz-conjecture/collatz-conjecture.ts
export function steps(number: number): number {
    if ((number <= 0) || (!Number.isInteger(number)) ) {
        throw new Error("Only positive integers are allowed");
    }

    let result: number = number;
    let steps: number = 0;

    while (result > 1) {
        steps += 1;
        if (result % 2 === 0) {
            result = Math.floor(result / 2);
        } else {
            result = 3 * result + 1;
        }
    }

    return steps;
}
